instance/models/server.py

```diff
@@ -461,15 +461,17 @@ def terminate(self):
         else:
             self._status_to_terminated()
 
-    def _shutdown(self, os_server, poll_interval=10, max_wait=settings.SHUTDOWN_TIMEOUT):
+    def _shutdown(self, os_server, poll_interval=10, max_wait=None):
         """
         Shutdown the server and wait to return until shutdown or wait threshold reached.
 
         We don't have an explicit state for this and don't catch exceptions, which is why this is private.
         The caller is expected to handle any exceptions and retry if necessary.
         """
+        max_wait = max_wait if max_wait is not None else settings.SHUTDOWN_TIMEOUT
```

Comment:
Might want to document that, I can imagine myself refactoring it if I didn't see the PR's context.

Reply:
@UmanShahzad, I'll add a comment. :+1:

Reply:
@UmanShahzad, I am not sure about adding a comment describing why testing the truthiness of `max_wait` shouldn't be done because it is just describing the way python works.

Reply:
Do the tests fail if this function disallows 0 with the refactoring I imagined above? If it does then it makes sense to not document. But if tests pass even if we perform the refactor, that could mean passing max_wait=0 in some places while not having intended effects, e.g. a hidden bug.

You can just leave this comment:

```python
# We purposely check for `None` rather than generically for falseness
# because it allows `max_wait = 0`.
```

Reply:
> Do the tests fail if this function disallows 0 with the refactoring I imagined above? 
@UmanShahzad , the tests don't fail but each test loops for 10 minutes (the default value for the `SHUTDOWN_TIMEOUT`) before continuing and that wastes a lot of time. I will add the suggested comment.

Reply:
Thanks!